src/Adapter/Category/CommandHandler/AddCategoryHandler.php

```diff
@@ -59,6 +59,6 @@ public function handle(AddCategoryCommand $command)
 
         $this->uploadImages($category, $command);
 
-        return new CategoryId($category->id);
+        return new CategoryId((int) $category->id);
```

Comment:
Why not casting in directly in CategoryId?

Reply:
see comments above. as i understand, it only validates that provided value is actual `int`, but does not fix it for you, as it may be bug.